aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/Orchestrator.java

```diff
@@ -288,91 +294,77 @@ private void stopObservingStorageChanges() {
     private void startApiSync() {
         LOG.info("Setting currentState to SYNC_VIA_API");
         currentState.set(State.SYNC_VIA_API);
-        disposables.add(startApiSyncCompletable()
-            .doOnComplete(() -> {
-                LOG.info("Started the orchestrator in API sync mode.");
-                publishReadyEvent();
+        disposables.add(
+            Completable.create(emitter -> {
+                LOG.info("Starting API synchronization mode.");
+
+                // Resolve any client provided DataStoreSyncExpressions, before starting sync and subscriptions, once
+                // each time DataStore starts.  The QueryPredicateProvider caches the resolved QueryPredicates, which
+                // are then used to filter data received from AppSync.
+                queryPredicateProvider.resolvePredicates();
+
+                subscriptionProcessor.startSubscriptions(this::transitionToLocalOnlyBlocking);
+
+                LOG.debug("About to hydrate...");
+                try {
+                    boolean subscribed = syncProcessor.hydrate()
+                            .blockingAwait(adjustedTimeoutSeconds, TimeUnit.SECONDS);
+                    if (!subscribed) {
+                        throw new TimeoutException("Timed out while performing initial model sync.");
+                    }
+                } catch (Throwable failure) {
+                    if (!emitter.isDisposed()) {
```

Comment:
😺

Reply:
This was already there, I just moved it a bit :). I believe the root cause though is that we didn't have a similar check for the `ReplaySubject buffer` before.  We did check `hasThrowable`, but I don't think that's quite the same.  So, the solution I've implemented is to just simplify and get rid of the buffer entirely.